src/html.ml

```diff
@@ -178,7 +254,36 @@ let rec block = function
       in
       elt Block "dl" attr (Some (concat_map f l))
 
-let of_doc doc = concat_map block doc
+let of_doc ?(auto_identifiers = true) doc =
+  let identifiers = Identifiers.empty in
+  let f identifiers = function
+    | Heading (attr, level, text) ->
+        let attr, identifiers =
+          if (not auto_identifiers) || List.mem_assoc "id" attr then
+            (attr, identifiers)
+          else
+            let id = slugify (to_plain_text text) in
+            (* Default identifier if empty. It matches what pandoc does. *)
+            let id = if id = "" then "section" else id in
+            let count, identifiers = Identifiers.touch id identifiers in
+            let id =
+              if count = 0 then id else Printf.sprintf "%s-%i" id count
+            in
```

Comment:
I feel like this might better belong in the `slugify` function. WDYT?

Reply:
I don't have any strong opinion on that one. I made the changes in [`820f310` (#267)](https://github.com/ocaml/omd/pull/267/commits/820f310bc9c7621aab419ba2c091968d1b954423)